core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -558,17 +574,38 @@ class ReplicaManager(val config: KafkaConfig,
         }
         partitionsToDelete += topicPartition
       }
+      if (stopPartition.deleteRemoteLog)
+        remotePartitionsToDelete += topicPartition
+
       // If we were the leader, we may have some operations still waiting for completion.
       // We force completion to prevent them from timing out.
       completeDelayedFetchOrProduceRequests(topicPartition)
     }
 
     // Third delete the logs and checkpoint.
     val errorMap = new mutable.HashMap[TopicPartition, Throwable]()
+    val remoteStorageErrorHandler = new BiConsumer[TopicPartition, Throwable] {
```

Comment:
Do we need this handler since it only does logging?

Reply:
Removed the `remoteStorageErrorHandler`